File: src/Models/Pile.swift
import Foundation

public struct Pile: Identifiable, Equatable, Codable {
    public enum PileType: String, Codable {
        case stock
        case waste
        case tableau
        case foundation
        case freeCell
    }
    
    public let id: String
    public let type: PileType
    public var cards: [Card]
    
    public init(id: String, type: PileType, cards: [Card] = []) {
        self.id = id
        self.type = type
        self.cards = cards
    }
    
    public var isEmpty: Bool {
        cards.isEmpty
    }
    
    public var topCard: Card? {
        cards.last
    }
}

File: android/app/src/main/java/com/leah/honeycomb/Pile.kt
package com.leah.honeycomb

import kotlinx.serialization.Serializable

@Serializable
enum class PileType {
    Stock,
    Waste,
    Tableau,
    Foundation,
    FreeCell
}

@Serializable
data class Pile(
    val id: String,
    val type: PileType,
    val cards: List<Card> = emptyList()
) {
    val isEmpty: Boolean get() = cards.isEmpty()
    
    val topCard: Card? get() = cards.lastOrNull()
}
